src/app/Fake.DotNet.Fsc/Fsc.fs

```diff
@@ -0,0 +1,480 @@
+        "-o" :: Path.changeExtension outExt (List.head srcFiles) :: opts @ srcFiles
+    |> Array.ofList
+
+/// Reports Fsc compile errors to the console using Fake.Core.Trace
+let private reportErrors (errors: FscResultMessage []) = 
+    for e in errors do
+        match e with
+        | FscResultMessage.Warning errMsg -> Trace.traceImportant errMsg
+        | FscResultMessage.Error errMsg -> Trace.traceError errMsg
+
+/// Compiles the given source files with the given options using either
+/// the internal FCS or an external fsc.exe. If no options
+/// given (i.e. the second argument is an empty list), by default tries
+/// to behave the same way as would the command-line 'fsc.exe' tool.
+let private compileFiles (compiler: CompilerFunc) (srcFiles : string list) (opts : string list) : int = 
+    let optsArr = makeArgsList opts srcFiles
+
+    Trace.trace <| sprintf "FSC with args:%A" optsArr
+    let errors, exitCode = compiler optsArr
+    
+    reportErrors errors
+    exitCode
+
+/// Common compiler arg prep code
+let private doCompile (compiler: CompilerFunc) (fscParams : FscParam list) (inputFiles : string list) : int = 
+    let inputFiles = inputFiles |> Seq.toList
+    let taskDesc = inputFiles |> String.separated ", "
+    let fscParams = if fscParams = [] then FscParam.Defaults else fscParams
+    let argList = fscParams |> List.map string
+
+    use __ = Trace.traceTask "Fsc " taskDesc
+    let res = compileFiles compiler inputFiles argList
+    res
+
+
+(*
+Compile using the internals of FCS
+*)
+/// The internal FCS Compiler
+let private scsCompile optsArr = 
+    let scs = FSharpChecker.Create()
+    // Always prepend "fsc.exe" since fsc compiler skips the first argument
+    let optsArr = Array.append [|"fsc.exe"|] optsArr
+    let errors, exitcode = scs.Compile optsArr |> Async.RunSynchronously
+
+    /// Better compile reporting thanks to:
+    /// https://github.com/jbtule/ComposableExtensions/blob/5b961b30668bb7f4d17238770869b5a884bc591f/tools/CompilerHelper.fsx#L233
+    let errors = errors |> Array.map (fun (e: FSharpErrorInfo) -> 
+        match e.Severity with
+            | FSharpErrorSeverity.Error -> FscResultMessage.Error e.Message
+            | FSharpErrorSeverity.Warning -> FscResultMessage.Warning e.Message)
+
+    errors, exitcode
+
+/// Compiles the given F# source files with the specified parameters.
+///
+/// ## Parameters
+///
+///  - `setParams` - Function used to overwrite the default Fsc parameters.
+///  - `inputFiles` - The F# input files.
+///
+/// ## Returns
+///
+/// The exit status code of the compile process.
+///
+/// ## Sample
+///
+///     ["file1.fs"; "file2.fs"]
+///     |> CompileWithResult [Out "" 
+///                 Target Exe
+///                 Platform AnyCpu
+///                 References []
+///                 Debug false 
+///             ]
+let CompileWithResult (fscParams : FscParam list) (inputFiles : string list) : int = 
```

Comment:
Can you please use lower-cased first letters for public APIs (these are our current api guidelines), see also `Compile`, `CompileExternalWithResult`, `CompileExternal`

Reply:
Oh gotcha, ok, sure thing!